docs/blog/2025-01-20-linking-ts-in-monorepos.md

```diff
@@ -0,0 +1,666 @@
+
+```
+ts-monorepo-linking
+   ├─ apps
+   │  └─ myapp
+   │     ├─ src
+   │     └─ tsconfig.json  <<< myapp TS program
+   ├─ packages
+   │  └─ lib-a
+   │     ├─ src
+   │     └─ tsconfig.json  <<< lib-a TS program
+   ├─ tsconfig.base.json
+   └─ tsconfig.json        <<< root-level solution tsconfig
+```
+
+The incremental nature of project references allows TypeScript to track changes and skip unnecessary recompilation, resulting in:
+
+- **Faster builds:** Only projects affected by changes are recompiled, saving time during development and on CI.
+- **Lower memory usage:** Processing smaller, isolated projects is more memory efficient, which is particularly helpful in large workspaces or resource-constrained environments like CI pipelines.
+- **Improved editor performance:** TypeScript's incremental setup ensures quicker type-checking and autocomplete, even in large monorepos.
+
+For a more deep dive check out our article on ... **TODO LINK ZACK's ARTICLE**
+
+## Combining TypeScript Project References and Package Manager Workspaces
+
+Example: [Stackblitz](https://stackblitz.com/github/juristr/ts-monorepo-linking/tree/workspaces-ts-proj-refs) - [Github](https://github.com/juristr/ts-monorepo-linking/tree/workspaces-ts-proj-refs)
+
+Modern package managers like NPM, PNPM, Yarn, and Bun have a so-called "workspaces feature" that allows for a more seamless resolution of local packages that allows for a more seamless resolution of local packages.
+
+{% tabs %}
+
+{% tab label="NPM/Yarn/Bun Workspaces" %}
+
+For most package managers, you can use the `workspaces` property in the root `package.json` to define the packages that are part of the monorepo.
+
+```json {% fileName="package.json" %}
+{
+  "name": "ts-monorepo-linking",
+  ...
+  "workspaces": [
+    "apps/*",
+    "packages/*"
+  ]
+}
+
+```
+
+{% /tab %}
+
+{% tab label="PNPM Workspaces" %}
+
+PNPM uses a `pnpm-workspace.yaml` file to define the packages that are part of the monorepo.
+
+```yaml {% fileName="pnpm-workspace.yaml" %}
+packages:
+  - 'apps/*'
+  - 'packages/*'
+```
+
+{% /tab %}
+{% /tabs %}
+
+This approach **eliminates the need for TypeScript path aliases for module resolution**.
+
+```diff {% fileName="tsconfig.base.json" %}
+{
+  "compilerOptions": {
+    "target": "ES2020",
+    "module": "NodeNext",
+    "strict": true,
+    "moduleResolution": "NodeNext",
+    "composite": true,
+    "declaration": true,
+    "declarationMap": true,
+    "sourceMap": true,
+-   "baseUrl": ".",
+-   "paths": {
+-     "@ts-monorepo-linking/lib-a": ["packages/lib-a/src/index.ts"]
+-   }
+  }
+}
+```
+
+Instead, the package manager's workspaces feature makes sure to link the packages properly s.t. they can be resolved correctly at build and runtime. This doesn't have any impact on our TypeScript project references setup which cares about type checking and resolves dependencies via the `references` property.
```

Comment:
```suggestion
Instead, the package manager's workspaces feature makes sure to link the packages properly such that they can be resolved correctly at build and runtime. This doesn't have any impact on our TypeScript project references setup which cares about type checking and resolves dependencies via the `references` property.
```

This is not a standard/ubiquitous abbreviation in English (not like e.g. or i.e.) so should be written out to avoid confusion